assets/js/googlesitekit/components-gm2/TextField.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * TextField component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import {
+	default as MaterialTextField,
+	HelperText,
+	Input,
+} from '@material/react-text-field';
+
+const TextField = ( {
+	className,
+	name,
+	label,
+	noLabel,
+	outlined,
+	textarea,
+	trailingIcon,
+	helperText,
+	helperTextPersistent,
+	id,
+	inputType,
+	value,
+	size,
+	maxLength,
+	tabIndex,
+	disabled,
+	onChange,
+	onKeyDown,
+} ) => {
+	return (
+		<MaterialTextField
+			className={ className }
+			name={ name }
+			label={ label }
```

Comment:
As mentioned below, without an `htmlFor`/some kind of association between this label and the text input, things aren't working as-expected here.

Reply:
Updated to use a fallback `id` using `useInstanceID`.